test/modal.test.js

```diff
@@ -20,38 +22,42 @@ describe('Modal', () => {
   })
 
   test('is shown by default', () => {
-    const wrapper = mount(<Modal onClose={noop} />)
-    expect(wrapper.find('.modal-content').exists()).toEqual(true)
+    const content = Date.now()
+    render(<Modal onClose={noop}><span>{content}</span></Modal>)
+    expect(screen.getByText(content)).toBeInTheDocument()
   })
 
   test('can be hidden/animated by manually passing isOpen', () => {
-    const wrapper = mount(<Modal isOpen={false} onClose={noop} />)
-    expect(wrapper.find('.modal-content').exists()).toEqual(false)
+    const content = Date.now()
+    render(<Modal isOpen={false} onClose={noop}><span>{content}</span></Modal>)
+    expect(screen.queryByText(content)).not.toBeInTheDocument()
   })
 
-  test('calls close handler when close button is clicked', () => {
+  test('calls close handler when close button is clicked', async () => {
     const onClose = jest.fn()
-    const wrapper = mount(<Modal onClose={onClose} />)
-    wrapper.find('.modal-close').simulate('click')
+    const user = userEvent.setup()
+    render(<Modal onClose={onClose} />)
+
+    await user.click(screen.getByRole('button', { name: 'Close Modal' }))
     expect(onClose).toHaveBeenCalled()
   })
 
   test('adds additional class string to default class', () => {
-    const wrapper = mount(<Modal onClose={noop} className="custom" />)
-    expect(wrapper.find('.modal-inner.custom').exists()).toEqual(true)
+    render(<Modal onClose={noop} className="custom" />)
+    expect(screen.getByRole('dialog')).toHaveClass('modal-inner', 'custom')
   })
 
   test('adds additional overlay class string to default overlay class', () => {
-    const wrapper = mount(
+    render(
       <Modal onClose={noop} overlayClassName="custom-overlay" />
     )
-    expect(wrapper.find('.modal-fade-screen.custom-overlay').exists()).toEqual(
-      true
-    )
+
+    const overlay = screen.getByRole('dialog').parentElement
```

Comment:
I don't love having to use `parentElement`, but that seemed to be the easiest option. Let me know if you have any ideas

Reply:
I think it makes sense in this case. We're talking about an overlay element, so it's naturally a wrapper to the modal. Another option would be to add a testId to the overlay element, but I don't think it would add any value here.